storage/azure/src/client/AzureClientStorage.ts

```diff
@@ -102,8 +114,30 @@ export class AzureClientStorage extends ClientStorage {
       assertRelativeDirectory(input.reference.relativeDirectory);
     if (typeof input.data === "string") await assertFileNotEmpty(input.data);
 
-    return this._clientWrapperFactory
+    return this._blockBlobClientWrapperFactory
       .create(input)
       .uploadInMultipleParts(input.data, input.options);
   }
+
+  public async deleteObject(input: AzureTransferConfigInput): Promise<void> {
```

Comment:
I added these functions only for testing the access function but I think they should have been implemented sooner (because they already exist on ServerStorage). So, I can definitely split the PR.

Reply:
There are a couple of issues I see with this:
1. It is broadening the API and we did not really have a production use case for this. If we add it to the API, somebody will use it, find bugs, we'll have to fix them, etc.
2. This is more blurry and maybe Karolis @Animagne could comment more but `ClientStorage` is mean for clients wro mostly read objects and sometimes upload them. They should not need to list everything that is in the directory, they should not delete objects. For use cases that they do (line checkpoints v2) they can request credentials from the server and then do whatever, but that is a very specific use case.